File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    result=[]
    set_lottos=set(lottos)
    set_win_nums=set(win_nums)
    left=list((set_lottos-set_win_nums))
    if 0 in left:
        left.remove(0)
    correct=list(set_lottos&set_win_nums)
    ll=len(left)
    lc=len(correct)
    if ll==6:
        ll-=1
    if lc==0:
        lc+=1
    return [ll+1,7-lc]

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.js
function solution(lottos, win_nums) {
    let score=[6,6,5,4,3,2,1];
    let correct=lottos.filter(item=>win_nums.includes(item)).length
    let z=lottos.filter(item=>item===0).length
    return [score[correct+z],score[correct]];
}